### PR TITLE
Feataure/update comments count with shared redux store

### PR DESCRIPTION
**Pull request by seoyoochan**
[Review on Reviewable](https://reviewable.io/reviews/shakacode/react-webpack-rails-tutorial/239)

### CODE REVIEW

**Comment by justin808** [comment]
Looking good. We want to see the shared store defined on the view for each of the router and non-router apps.

---

Reviewed 4 of 7 files at r1, 10 of 10 files at r2.
Review status: 13 of 14 files reviewed at latest revision, 9 unresolved discussions.

---

*[app/assets/javascripts/rails_startup.js, line 5 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw6RknvKCDdD5_gzr0:-KBw6RknvKCDdD5_gzr1:1934884044) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/5b8787152b96e60595802ab1091089003d5b0f09/app/assets/javascripts/rails_startup.js#L5)):*
@seoyoochan Just checking we  didn't lose the existing highlight functionality. This code sets the active class.

---

*[app/controllers/comments_controller.rb, line 73 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw6eqr6I2txEuB1Y32:-KBw6eqr6I2txEuB1Y33:-203794074) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/app/controllers/comments_controller.rb#L73)):*
we don't need a separate count -- we can get the count from the comments object: comments.length

---

*[app/controllers/pages_controller.rb, line 39 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw6lvM5Mj3xbsuuZud:-KBw6lvM5Mj3xbsuuZue:-927236912) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/app/controllers/pages_controller.rb#L39)):*
we don't need a separate count -- we can get the count from the comments object: comments.length

Also, no reason to sort if you wanted the size., but you could be trying to cache? In any case you duplicated the code.

---

*[app/views/layouts/application.html.erb, line 22 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw7-105iLbbwXccnff:-KBw7-105iLbbwXccnfg:870030568) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/app/views/layouts/application.html.erb#L22)):*
There's no props. This should come from the shared redux store. I'm not sure about the pages without the shared store. 

Use this API:

```js

  /**
   * Get an Object containing all hydrated stores. Useful for debugging.
   * @returns {*}
   */
  stores() {
    return StoreRegistry.stores();
  },
```

so you can check if the store exists before calling get store, which will crash if the store doesn't exist

if no store, don't display a count.

---

*[client/package.json, line 78 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw7jRyOU7zbUN0UFuK:-KBw7jRzeMDpfo9RhBm4:-55489011) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/client/package.json#L78)):*
we can update the dependencies if you like

---

*[client/app/bundles/comments/containers/NonRouterCommentsContainer.jsx, line 26 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw7qQu56v515HpyjId:-KBw7qQu56v515HpyjIe:-1280936503) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/client/app/bundles/comments/containers/NonRouterCommentsContainer.jsx#L26)):*
remove

---

*[client/app/bundles/comments/containers/RouterCommentsContainer.jsx, line 5 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw7uEB55IwIGTsfnvU:-KBw7uEB55IwIGTsfnvV:-1280936503) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/client/app/bundles/comments/containers/RouterCommentsContainer.jsx#L5)):*
remove

---

*[client/app/bundles/comments/containers/RouterCommentsContainer.jsx, line 31 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw7w9RnXP8pUsQrgfh:-KBw7w9RnXP8pUsQrgfi:-1280936503) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/client/app/bundles/comments/containers/RouterCommentsContainer.jsx#L31)):*
remove

---

*[client/app/bundles/comments/startup/clientRegistration.jsx, line 22 \[r2\]](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-KBw8ARcOyRE7aSGApeo:-KBw8ARcOyRE7aSGApep:-829954617) ([raw file](https://github.com/shakacode/react-webpack-rails-tutorial/blob/fdb2777ba6bd2a870e370541cab4f65ccaf20dcc/client/app/bundles/comments/startup/clientRegistration.jsx#L22)):*
This is a little tricky.

Can we use the same "shareCommentsStore" for both the router and non-router apps?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-:-KBw8POeEFq8AXnNgOEn:699315371)*

**Comment by justin808** [comment]
@seoyoochan Can we wrap this one up today?

**Comment by seoyoochan** [comment]
@justin808 just pushed commit. 
[fixed and cleaned up store logics, display comments count!](https://github.com/shakacode/react-webpack-rails-tutorial/pull/239/commits/546ab76481ed67a69b89cb711704fd0d68ac5c44)

**Comment by justin808** [comment]
Not a single change! :100: 

We just do one more commit for the revised API!

---

Reviewed 1 of 7 files at r1, 4 of 15 files at r4, 13 of 14 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-:-KD2YA0eUVEb_7LWTLS4:-200769799)*

**Comment by justin808** [comment]
Reviewed 2 of 3 files at r7, 6 of 6 files at r8.
Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239)*

**Comment by justin808** [comment]
Almost there!

---

Review status: all files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/shakacode/react-webpack-rails-tutorial/239#-:-KD31TqIWGMGC6LgX7-5:-810123414)*

File: client/app/bundles/comments/components/Navigationbar/Navigationbar.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+              </li>
+              <li>
+                <a href={
+                  'http://forum.shakacode.com/' +
+                  't/fast-rich-client-rails-development' +
+                  '-with-webpack-and-the-es6-transpiler/82/22'
+               }>Forum</a>
+              </li>
+              <li>
+                <a id='js-comment-count'
```
Comment (robwise) [inline client/app/bundles/comments/components/Navigationbar/Navigationbar.jsx]: hide this if we don't have a comments count

**Comment by justin808** [comment]
Moved to #254

**Pull request closed without merging**